### PR TITLE
chore(framework): Continue to move loaders to framework

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
More move and cleanup

FIXES FRMW-2603
FIXES FRMW-2608
FIXES FRMW-2610
FIXES FRMW-2611

### CODE REVIEW

File: packages/medusa/package.json
Comment (adrien2p) [inline packages/medusa/package.json]: I ll have a disussion with @olivermrbl on this issue I am having at the moment

**Comment by adrien2p** [comment]
@olivermrbl once this pr is ok (meaning only the tests relying on the CLI/starter fail) I will revert the version to preview, then everything will fail. Should we still merge after that change?

File: packages/medusa/package.json
Comment (adrien2p) [inline packages/medusa/package.json]: Only apply this change before merging

```suggestion
    "@medusajs/framework": "^0.0.1-preview-20240724081425",
```

File: integration-tests/http/package.json
Comment (sradevski) [inline integration-tests/http/package.json]: I guess this isn't intentional :) If it was about the flaky test, it should be fixed now, but let me know if it happens again.

**Review by sradevski — APPROVED** [review]
LGTM!

**Comment by adrien2p** [comment]
@olivermrbl should we apply the comment changes and merge? Wdyt? I already have another branch from this one for the next part

File: integration-tests/http/package.json
Comment (adrien2p) [reply integration-tests/http/package.json]: Definitely not 👌 thanks I missed it in my review

**Review by olivermrbl — APPROVED** [review]

**Comment by olivermrbl** [comment]
@adrien2p, let's do it!

**Comment by adrien2p** [comment]
@olivermrbl just merged the revert of the version, I expect the tests to fail but I think it is temporary right?

**Comment by olivermrbl** [comment]
@adrien2p, I believe the lock-file should also be updated before we merge right?

**Comment by adrien2p** [comment]
> @adrien2p, I believe the lock-file should also be updated before we merge right?

Yes thats correct

**Comment by olivermrbl** [comment]
@adrien2p, just playing around with making the framework-versions work. I'll revert my commits if I can't solve it.

**Comment by adrien2p** [comment]
Got it 👍 no problem at all let me know now how it goes and the solution

**Comment by olivermrbl** [comment]
> let me know now how it goes and the solution

Do you think the latest changes could work?

**Comment by adrien2p** [comment]
@olivermrbl I believe after this last commit we should be able to finally merge it

**Comment by adrien2p** [comment]
@olivermrbl should we merge as it is?

File: packages/framework/framework/package.json
Comment (adrien2p) [inline packages/framework/framework/package.json]: @olivermrbl should i move that back to preview before we merge?

Comment (olivermrbl) [reply packages/framework/framework/package.json]: Can you move them to use their latest versions with a caret (`^`). I think then we can merge

Comment (adrien2p) [reply packages/framework/framework/package.json]: let me try, last time it was failing for me 😂

Comment (adrien2p) [reply packages/framework/framework/package.json]: Looks like this breaks everything 🤒

Comment (adrien2p) [reply packages/framework/framework/package.json]: let me try something

Comment (olivermrbl) [reply packages/framework/framework/package.json]: ~~Think the wrong version of utils is installed right now. There is no 1.11.10~~

Comment (adrien2p) [reply packages/framework/framework/package.json]: Yes there is one https://www.npmjs.com/package/@medusajs/utils/v/1.11.10

Comment (adrien2p) [reply packages/framework/framework/package.json]: Otherwise the install would have failed with 404 and not reaching the build

Comment (adrien2p) [reply packages/framework/framework/package.json]: Now I have another question, how come 1.11.10 does not have the latest changes when it is the latest version and i have to downgrade ?

Comment (olivermrbl) [reply packages/framework/framework/package.json]: Yes, I noticed this. Edited my comment :)

Comment (olivermrbl) [reply packages/framework/framework/package.json]: > Now I have another question, how come 1.11.10 does not have the latest changes when it is the latest version and i have to downgrade ?

This is what spawned my comment initially. I was wondering this too. My immediate thought was that this is because 1.11.10 has been released as part of v1.x, and that branch doesn't have everything from `develop`.

Comment (adrien2p) [reply packages/framework/framework/package.json]: ahah makes sense 👍

Comment (olivermrbl) [reply packages/framework/framework/package.json]: Pretty sure this is actually what has been causing issues all along.

Comment (adrien2p) [reply packages/framework/framework/package.json]: Yes probably, and i ve missed it for so long 😣

Comment (olivermrbl) [reply packages/framework/framework/package.json]: Let's slap an automerge on this, shall we?

Comment (adrien2p) [reply packages/framework/framework/package.json]: done 😍

**Pull request merged**
